### PR TITLE
Add webhooks for use with document upsert

### PR DESCRIPTION
**Pull request by Abdullah13521**
## Description

Addresses [this](https://github.com/tjmlabs/ColiVara/issues/38) issue

## Type of change

- [ ] 🚨 Breaking change - causes existing functionality to break (triggers MAJOR version bump)
- [x] ✨ New feature - adds functionality in a non-breaking way (triggers MINOR version bump)
- [ ] 🐛 Bug fix - fixes an issue without changing functionality (triggers PATCH version bump)
- [ ] 🏠 Internal - changes that don't affect external functionality (no version bump)
      (e.g., refactoring, docs, tests, admin site changes)

## Release Impact

- Current version: 
- Next version will be: 

## Checklist

- [x] My code follows the style guidelines 
- [x] My code is tested with 99%+ coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have updated the documentation to reflect my changes
- [x] I have added/updated tests to cover my changes

### CODE REVIEW

File: web/api/tests/tests.py [hunk 1]
```diff
@@ -309,6 +309,33 @@ async def test_delete_collection_not_found(async_client, user, collection):
 """ Document tests """
 
 
+async def test_add_webhook(async_client, user):
```
Comment (Abdullah13521) [inline web/api/tests/tests.py:321]: I assume we need to add this to the sdk, right?

File: web/api/views.py [hunk 2]
```diff
@@ -410,6 +470,27 @@ async def process_upsert_document(
             )
         )
         logger.info(f"Document {document.name} processed successfully.")
+
+        if not payload.wait and request.auth.svix_application_id:
```
Comment (Abdullah13521) [inline web/api/views.py]: should we make the webhook event consume credits in the cloud branch?

File: web/api/tests/tests.py (hunk 1 above)
Comment (Jonathan-Adly) [reply web/api/tests/tests.py:321]: yes - I saw the PR there. It is good!

File: web/api/views.py (hunk 2 above)
Comment (Jonathan-Adly) [reply web/api/views.py]: The credits consumed would be on the number of pages (no change). The webhook itself, addition, deletion is wouldn't affect credits.

**Comment by Jonathan-Adly** [comment]
This looks good Abdullah. Just to keep track. To do:

- Add env in settings w/ a None default.

- Return clear errors when someone tries to add a webhook, but didn't setup a SVIX token

- Write docs on how can the user set it up via the API or the dashboard on cloud 

- Make sure SDK is setup to add the url via API and to do signature verification to ingest the webhook. 

- Setup the Svix dashboard for production 

(We merge to main and cloud here)

- Add link/form to edit-account/ on cloud so users can add their webhook URL
- If they have one - then display it, with option to change.
- Via a short snippet explain email vs. webhook behavior and link to docs.

(We merge to cloud only here)

**Review by Jonathan-Adly — APPROVED** [review]

**Comment by Jonathan-Adly** [comment]
I will change the url from documents/webhook/ to just /webhook/. 

Since there is documents/<str:document_name>, this can be misinterpreted as document. I will updated the SDK as well,

**Pull request merged**